core/services/registrysyncer/capabilities_registry_reader.go

```diff
@@ -0,0 +1,271 @@
+package registrysyncer
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	p2ptypes "github.com/smartcontractkit/libocr/ragep2p/types"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/types"
+	kcr "github.com/smartcontractkit/chainlink-evm/gethwrappers/keystone/generated/capabilities_registry_1_1_0"
+	kcrv2 "github.com/smartcontractkit/chainlink-evm/gethwrappers/workflow/generated/capabilities_registry_wrapper_v2"
+	"github.com/smartcontractkit/chainlink/v2/core/services/relay/evm/capabilities/versioning"
+	evmrelaytypes "github.com/smartcontractkit/chainlink/v2/core/services/relay/evm/types"
+)
+
+// Error constants for unsupported operations
+var (
+	ErrNotSupportedInV1 = errors.New("operation not supported in V1 capabilities registry")
+)
+
+// CapabilitiesRegistryReader defines the interface for reading from capabilities registry contracts
+// across different versions. This interface abstracts the version-specific differences in the
+// contract structure and provides a unified way to access capabilities, DONs, and nodes.
+type CapabilitiesRegistryReader interface {
+	// Core methods supported by all versions
+	GetCapabilities(ctx context.Context) ([]CapabilityInfo, error)
+	GetDONs(ctx context.Context) ([]DONInfo, error)
+	GetNodes(ctx context.Context) ([]NodeInfo, error)
+	Address() common.Address
+	Close() error
+
+	// V2-specific methods (return ErrNotSupportedInV1 for V1 registries)
+	GetDONsInFamily(ctx context.Context, donFamily string) ([]uint32, error)
+	GetHistoricalDONInfo(ctx context.Context, donID uint32, configCount uint32) (*DONInfo, error)
+	GetNode(ctx context.Context, p2pID [32]byte) (*NodeInfo, error)
+	GetNodeOperator(ctx context.Context, nodeOperatorID uint32) (*NodeOperator, error)
+	GetNodeOperators(ctx context.Context) ([]NodeOperator, error)
+	GetNodesByP2PIds(ctx context.Context, p2pIDs [][32]byte) ([]NodeInfo, error)
+	IsCapabilityDeprecated(ctx context.Context, capabilityID string) (bool, error)
+}
+
+// CapabilityInfo represents capability information across all versions
+// Version-specific fields are pointers and will be nil for versions that don't support them
+type CapabilityInfo struct {
+	// Common fields across all versions
+	ID                    string
+	LabelledName          string
+	Version               string
+	CapabilityType        uint8
+	ResponseType          uint8
+	ConfigurationContract common.Address
+	IsDeprecated          bool
+
+	// V1-specific fields
+	HashedID *[32]byte `json:"hashedId,omitempty"`
+
+	// V2-specific fields
+	Metadata *[]byte `json:"metadata,omitempty"`
+}
+
+// DONInfo represents DON information across all versions
+// Version-specific fields are pointers and will be nil for versions that don't support them
+type DONInfo struct {
```

Comment:
copy pasta to a v2 package and add the needed fields.  If you want the v2 can import the DONInfo from maybe a sibling package.  No need to add Version indicator